amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/validation/PayloadsInApiCollector.scala

```diff
@@ -30,79 +30,98 @@ class PayloadsInApiCollector(model: BaseUnit) {
+        collectFromNewShape(candidateCollector, shapesCollector, union)
+        union.anyOf
+          .foreach(member => shouldSkip += member.id) // Skip generate candidates for union members, only union
       case shape: AnyShape =>
-        val examples = shape.examples.collect({
-          case example: Example
-              if example.fields.exists(ExampleModel.StructuredValue) && example.strict.option().getOrElse(true) =>
-            DataNodeCollectedElement(example.structuredValue, example.id, example.raw.value(), example.annotations)
-          case example: Example if example.fields.exists(ExampleModel.Raw) && example.strict.option().getOrElse(true) =>
-            StringCollectedElement(example.id, example.raw.value(), example.annotations)
-        })
-        if (examples.nonEmpty) {
-          results.put(shape.id, examples ++ getDefault(shape))
-          shapes.put(shape.id, shape)
-        } else {
-          getDefault(shape) match {
-            case Some(ei) =>
-              results.put(shape.id, Seq(ei))
-              shapes.put(shape.id, shape)
-            // first time i check the shape, i should collect it if has values
-            case None if shape.values.nonEmpty =>
-              shapes.put(shape.id, shape)
-              results.put(shape.id, Nil)
-            case _ => // ignore
-
-          }
-        }
-
+        collectFromNewShape(candidateCollector, shapesCollector, shape)
       case _ =>
     }
-    val seq = results
-      .flatMap({
-        case (id, e) =>
-          val shape = shapes(id)
-          if (e.isEmpty) // only shapes with enums
-            Seq(
-              ValidationCandidate(
-                shape,
-                PayloadFragment().withMediaType(ShapesNodesValidator.defaultMediaTypeFor(shape.values.head))))
-          else
-            e.map { encodes =>
-              ValidationCandidate(shape, buildFragment(shape, encodes))
-            }
-      })
-      .toSeq
-    seq
+
+    candidateCollector.flatMap {
+      case (id, e) if !shouldSkip.contains(id) =>
+        val shape = shapesCollector(id)
+
+        val enumMemberCandidates = shape.values.map { enumMember =>
+          val fragment = PayloadFragment().withMediaType(ShapesNodesValidator.defaultMediaTypeFor(enumMember))
+          ValidationCandidate(shape, fragment)
+        }
+
+        val examplesCandidates = e.map { encodes =>
+          ValidationCandidate(shape, buildFragment(shape, encodes))
+        }
+
+        enumMemberCandidates ++ examplesCandidates
+
+      case _ => Nil
+    }.toSeq
   }
 
+  private def collectFromNewShape(results: mutable.Map[String, Seq[CollectedElement]],
+                                  shapes: mutable.Map[String, Shape],
+                                  shape: AnyShape) = {
+    val examples = shape.examples.collect({
+      case example: Example
+          if example.fields.exists(ExampleModel.StructuredValue) && example.strict.option().getOrElse(true) =>
```

Comment:
[OOS] - Condition is similar to the one below (not the same though). Is there any domain significance this condition has? Could be the name of the extracted variable/function